infrastructure/smart-contracts/contracts/xai-subsidy/ClientDocs.md

```diff
@@ -0,0 +1,207 @@
+	const nonce = await forwarderContract.methods.nonces(userWallet).call();
+	return Number(nonce);
+}
+```
+
+Now to sign this typedData we use the EIP-712 standard to get the signature. We use the signTypedData() from wagmi/core:
+
+```javascript
+const signature = await signTypedData(config, typedData);
+```
+
+The config in this example is our defaultWagmiConfig.
+
+## How to forward the meta-transaction to the XAI Relayer API
+
+Use the signed request signature and the raw request body to forward the transaction and get the transaction fee sponsored by the XAI Gas Subsidy:
+
+```javascript
+type ForwardRequest = {
+    from: string;
+    to: string;
+    value: string;
+    gas: string;
+    nonce: string;
+    data: string;
+    signature: string;
+    forwarderAddress: string;
+};
+
+const forwardTransaction = async (
+    {
+        typedData, // Typed data from createTypedData()
+        signature, // user signature from signTypedData()
+        forwarderAddress // The forwarder address used
+    }:
+    {
+        typedData: TypedData,
+        signature: string,
+        forwarderAddress: string
+    }
+) => {
+
+    //Create the request body from the meta-transaction object the user signed
+    const requestBody: ForwardRequest = {
+        from: typedData.message.from,
+        to: typedData.message.to,
+        value: "0",
+        gas: typedData.message.gas,
+        nonce: typedData.message.nonce,
+        data: typedData.message.data,
+        signature,
+        forwarderAddress
+    }
+
+    try {
+        const res = await fetch(`https://relayer.xai.games/forward/[YOUR_PROJECT_ID]`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(requestBody),
+        });
+
+        const data = await res.json();
+        if (!res.ok) {
+            throw new Error(data.message);
+        }
+        console.log("Transaction complete: ", data.txHash)
+
+    } catch (error) {
+        console.error('Error fetching data:', error);
+        alert("Error sending to relayer\n" + error);
+    }
+}
+```
+
+The request will return the transaction hash as a success response or an error string if the transaction would fail or the user has not enough balance left for their sponsoring with your project.
+
+You can request the users current balance from a GET request to the XAI Relayer API `https://relayer.xai.games/quota/[YOUR_PROJECT_ID]/[USER_WALLET]`
+This will return the user's quota Object:
+
+```javascript
+{
+    balanceWei: string;             // The remaining amount for a user that you pay for his transaction gas
+    nextRefillTimestamp: number;    // The next moment the balance gets refilled
+    nextRefillAmountWei: string;    // The amount of wei that gets refilled on the next refill
+    lastRefillTimestamp: number;    // The last time the balance for a user got refilled
+}
+```
+
+## Get ProjectID
```

Comment:
Should this be moved to the top? Step 1, request your project Id?

Reply:
I think the flow will make them get in tough before looking at the docs, so will will already provide them.

I have cut this section to 

Get a Xai Subsidy ProjectID
Contact us to get your ProjectID

Just for reference